gulp/createskin.js

```diff
@@ -0,0 +1,206 @@
+     */
+    function validate(value, regex, name, len) {
+        if (value === '') {
+            return 'Please provide a valid value';
+        }
+        var cases = value.match(regex);
+        if (cases === null) {
+            return 'Invalid value, please try again.';
+        }
+        if (value.length > len) {
+            return name + ' too long. Please use '+len+' chars or less';
+        }
+        return true;
+    }
+
+    /**
+     * Exports core plugin skins for sdk if core plugins are not already downloaded.
+     */
+    function downloadCorePlugins() {
+
+        var putils = require('../lib/plugin-utils');
+        var server = require('../lib/plugin-server.js')(gulp, gutil).getServer();
+        if (fs.existsSync(server.coreOutputDir())) {
+            if (gutil.env['debug']) {
+                putils.logDebug(gutil, "Core plugins exist in directory " + server.coreOutputDir() + "/ . Skipping download...");
+            }
+            //Start skin creation process
+            return createNewSkin();
+        }
+        var stream = through().obj();
+        var pluginExport = require('../lib/plugin-export.js')(gulp, gutil);
+        //Download core plugins
+        pluginExport.exportPlugin(server, {
+            pluginType: 'core',
+            doClear: false,
+            verboseMode: gutil.env['verbose'],
+            debugMode: gutil.env['debug'],
+            coreOutputDir: server.coreOutputDir()
+        }, undefined, function(err) {
+            if (err) {
+                putils.logError(gutil, "Error downloading core plugins! Error: "+err.message);
+                process.exit(1);
+            }
+            //Start skin creation process
+            return createNewSkin();
+        }).pipe(stream);
+    }
+
+    /**
+     * Interactive create skin function. Allows to create a brand new skin with responsive and non responsive support. 
+     * SDK skins are also valid skin parents. Creates a blank skin directory structure under skin base dir of sdk project.
+     */
+    function createNewSkin() {
+        var skinLib = require('../lib/skin')(gulp, gutil);
+        var putils = require('../lib/plugin-utils');
+
+        //Check if we are creating skin on a sdk project
+        if (!fs.existsSync('./server.conf.json')) {
+            console.log(gutil.colors.red('Please run create-skin under your Lithium SDK project directory.' +
+                '\n or create a new project with ' + gutil.colors.bold('li create-project') + ' before creating a new skin.'));
+            process.exit(1);
+            return true;
+        }
+
+        var options = {
+            skinInfo: {}
+        };
+
+        inquirer.prompt([
+            {
+                type: 'input',
+                name: 'name',
+                message: 'What is your skin ID?\nOnly lower case characters and numbers are allowed.',
+                validate: function (val) {
```

Comment:
May be we should emulate how studio works? Just ask for skin name and convert it into an id in the backend.